### PR TITLE
[TASK] Use signal afterExtensionInstall instead of hasInstalledExtensions (#1139)

### PR DESCRIPTION
**Pull request by tomasnorre**
Resolves: #1139

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -10,3 +10,7 @@ parameters:
 			count: 1
 			path: src/Set/Typo3RectorSetProvider.php
 
+		-
```
Comment (tomasnorre) [inline phpstan-baseline.neon:13]: I had to add this to have it pass the PHPStan. I could not figure out how to assign the value to the ` $node->args[1]->value` without using the `$this->createArg()` 

https://github.com/sabbelasichon/typo3-rector/pull/1624/files#diff-42ad5bdb72bdf2ba6a8115a562bc10ea4a17046fdea25bb65a9c2973616bd997R48

Comment (tomasnorre) [reply phpstan-baseline.neon:13]: I found this as an option
```php
$node->args[1]->value = new Node\Arg(BuilderHelpers::normalizeValue('afterExtensionInstall'));
```
but the `BuilderHelps` is marked as `@internal`

Comment (sabbelasichon) [reply phpstan-baseline.neon:13]: @tomasnorre Either you assign the new Arg this way: $node->args[1] = $this->createArg('afterExtensionInstall'). Behind the scenes Rector is using the BuilderHelpers::normalizeValue method or you assign a new value that way: $node->args[1]->value = new Node\Scalar\String_('afterExtensionInstall'); You can also use the BuilderHelpers::normalizeValue('afterExtensionInstall') method. 

Afterwards you can remove the line in the phpstan-baseline.neon.

**Comment by sabbelasichon** [comment]
@tomasnorre Really good for the first one. Keep it up.

File: phpstan-baseline.neon (hunk 1 above)
Comment (tomasnorre) [reply phpstan-baseline.neon]: I liked this option the most, as you see that it's `$node->args[1]->value` being set, but PHPStan complains about partials use: 
`$node->args[1]->value = new Node\Scalar\String_('afterExtensionInstall');`

So I went for ` $node->args[1] = $this->createArg('afterExtensionInstall')` instead.

**Pull request merged**